lib/experimental/register-webfonts-from-theme-json.php

```diff
@@ -5,170 +5,66 @@
  * @package gutenberg
  */
 
-/**
- * Register webfonts defined in theme.json.
- */
-function gutenberg_register_webfonts_from_theme_json() {
-	// Get settings.
-	$settings = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data()->get_settings();
-
-	// If in the editor, add webfonts defined in variations.
-	if ( is_admin() || ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ) {
-		$variations = WP_Theme_JSON_Resolver_Gutenberg::get_style_variations();
-
-		foreach ( $variations as $variation ) {
-
-			// Sanity check: Skip if fontFamilies are not defined in the variation.
-			if (
-				empty( $variation['settings'] ) ||
-				empty( $variation['settings']['typography'] ) ||
-				empty( $variation['settings']['typography']['fontFamilies'] )
-			) {
-				continue;
-			}
-
-			// Merge the variation settings with the global settings.
-			$settings['typography']                 = empty( $settings['typography'] ) ? array() : $settings['typography'];
-			$settings['typography']['fontFamilies'] = empty( $settings['typography']['fontFamilies'] ) ? array() : $settings['typography']['fontFamilies'];
-			$settings['typography']['fontFamilies'] = array_merge( $settings['typography']['fontFamilies'], $variation['settings']['typography']['fontFamilies'] );
-
-			// Make sure there are no duplicates.
-			$settings['typography']['fontFamilies'] = array_unique( $settings['typography']['fontFamilies'] );
-		}
-	}
-
-	// Bail out early if there are no settings for webfonts.
-	if ( empty( $settings['typography'] ) || empty( $settings['typography']['fontFamilies'] ) ) {
-		return;
-	}
-
-	$webfonts = array();
-
-	// Look for fontFamilies.
-	foreach ( $settings['typography']['fontFamilies'] as $font_families ) {
-		foreach ( $font_families as $font_family ) {
-
-			// Skip if fontFace is not defined.
-			if ( empty( $font_family['fontFace'] ) ) {
-				continue;
-			}
-
-			$font_family['fontFace'] = (array) $font_family['fontFace'];
-
-			foreach ( $font_family['fontFace'] as $font_face ) {
-				// Skip if the webfont was registered through the Webfonts API.
-				if ( isset( $font_face['origin'] ) && 'gutenberg_wp_webfonts_api' === $font_face['origin'] ) {
+if ( ! function_exists( '_wp_register_webfonts_from_theme_json' ) ) {
+	/**
+	 * Register webfonts defined in theme.json
+	 *
+	 * @param array $settings The theme.json file.
+	 */
+	function _wp_register_webfonts_from_theme_json( $settings ) {
+		// If in the editor, add webfonts defined in variations.
+		if ( is_admin() || ( defined( 'REST_REQUEST' ) && REST_REQUEST ) ) {
+			$variations = WP_Theme_JSON_Resolver_Gutenberg::get_style_variations();
+
+			foreach ( $variations as $variation ) {
+
+				// Sanity check: Skip if fontFamilies are not defined in the variation.
+				if (
+					empty( $variation['settings'] ) ||
+					empty( $variation['settings']['typography'] ) ||
+					empty( $variation['settings']['typography']['fontFamilies'] )
+				) {
```

Comment:
```suggestion
				if ( empty( $variation['settings']['typography']['fontFamilies'] ) ) {
```

Empty runs on each level so no need to do so manually, see https://3v4l.org/uItu1

Reply:
🤯 I didn't know that... beautiful! 👍